src/Microsoft.DocAsCode.Build.Engine/LinkPhaseHandlerWithIncremental.cs

```diff
@@ -65,7 +65,14 @@ private void PreHandle(List<HostService> hostServices)
         {
             using (new LoggerPhaseScope("ReloadModelsPerChanges", LogLevel.Verbose))
             {
-                ReloadModelsPerChanges(hostServices);
+                if (IncrementalContext.IsTemplateUpdated)
+                {
+                    ReloadModels(hostServices);
```

Comment:
for perf consideration. if mustache template is updated, no need to fully build.